packages/core/register/src/hook.js

```diff
@@ -0,0 +1,77 @@
+import Parcel from '@parcel/core';
+import {addHook} from 'pirates';
+import process from 'process';
+import path from 'path';
+import fs from 'fs';
+import syncPromise from '@parcel/utils/lib/syncPromise';
+
+let revert = null;
+const DEFAULT_CLI_OPTS = {
+  watch: false
+};
+
+// The filehandler returns a function that transforms the code
+function fileHandler(opts) {
+  let parcel = new Parcel({
+    entries: [path.join(process.cwd(), 'index.js')],
+    cliOpts: opts
+  });
+
+  return function(code, filename) {
+    let isInMonorepo =
+      filename.includes('parcel/packages') &&
+      !filename.includes('core/register');
+    if (filename.includes('@parcel') || isInMonorepo) {
+      return code;
+    }
+
+    try {
+      let result = syncPromise(
```

Comment:
Not sure if we should keep deasync in parcel

Reply:
haha well... idk how else you could do it if node requires hooks to be synchronous.